content/en/continuous_integration/guides/pipeline_data_model.md

```diff
@@ -0,0 +1,88 @@
+
+In the public API endpoint, you can populate the `previous_attempt` field to link to previous retries. Retries are treated as separate pipeline executions in Datadog, and the start and end time should only encompass that retry.
+
+### Partial retries
+
+When retrying a subset of jobs within a pipeline, you must send a new pipeline event with a new pipeline unique ID. The payload for any new jobs must be linked to the new pipeline unique ID. To link them to the previous retry, you can provide the `previous_attempt` field. Partial retries are treated as separate pipelines as well. The start and end time must not include the time of the original retry. For a partial retry, do not send payloads for jobs that ran in the previous attempt. Also, set the `partial_retry` field to `true` on partial retries to exclude them from aggregation when calculating run times.
+
+For example, a pipeline named `P` has three different jobs, namely `J1`, `J2`, and `J3`, executed sequentially. In the first run of `P`, only `J1` and `J2` are executed, and `J2` fails. Therefore, you need to send a total of three payloads:
+
+1. Job payload for `J1`, with ID `J1_1` and pipeline ID `P_1`.
+2. Job payload for `J2`, with ID `J2_1` and pipeline ID `P_1`.
+3. Pipeline payload for `P`, with ID `P_1`.
+
+Now, suppose there is a partial retry of the pipeline starting from `J2`, where all the remaining jobs succeed. You need to send three additional payloads:
+
+1. Job payload for `J2`, with ID `J2_2` and pipeline ID `P_2`.
+2. Job payload for `J3`, with ID `J3_1` and pipeline ID `P_2`.
+3. Pipeline payload for `P`, with ID `P_2`.
+
+The actual values of the IDs are not important. What matters is that they are correctly modified based on the pipeline run, as specified in the paragraph above.
+
+### Blocked pipelines
+
+If a pipeline is indefinitely blocked due to requiring manual intervention, a pipeline event payload must be sent as soon as the pipeline reaches the blocked state. The pipeline status must be set to `blocked`. The remaining pipeline data must be sent in separate payloads with a different pipeline unique ID. This second pipeline should have `is_resumed` set to `true`.
+
+{{< img src="ci/ci-pipeline-blocked-pipelines-execution.png" alt="Flow of a blocked pipeline execution" style="width:90%;">}}
+
+### Downstream pipelines
+
+If a pipeline is triggered as a child of another pipeline, the `parent_pipeline` field should be populated with details of the parent pipeline.
+
+### Manual pipelines
+
+If a pipeline is triggered manually, the `is_manual` field must be set to true.
+
+## Git information
+
+Payloads must contain Git information as specified in the public API endpoint [specification][3], if it can be retrieved.
+
+Sometimes Git information is not accessible, for example, when the CI provider cannot access the Git information on manually triggered pipelines. In these cases the `user` field can be populated instead with the user name and email.
```

Comment:
This was something we did for Codefresh but we have decided in the end to not use this, right @kassenq ? I would stress more that the Git information is almost required.

Reply:
Yes, currently RevEng team only looks at git commit author emails - to be billed on something that is passed into the `user` field we, we have to validate the user input there and move it to the git author email field they use for billing. Providing this field for Codefresh was more of a workaround. I would remove these lines for now

Reply:
Should we remove the `user` field from the public API specification then?